### PR TITLE
SAK-49309 CKEditor check for an undefined portal var

### PR DESCRIPTION
**Pull request by danielmerino**

### CODE REVIEW

File: library/src/webapp-filtered/editor/ckeditor.launch.js [hunk 1]
```diff
@@ -194,7 +194,7 @@ sakai.editor.editors.ckeditor.launch = function(targetId, config, w, h) {
 
     let siteId = "";
     if (collectionId) {
-        if (collectionId.startsWith('/user/') && portal && portal.siteId) {
+        if (collectionId.startsWith('/user/') && typeof portal != 'undefined' && portal.siteId) {
```
Comment (ottenhoff) [inline library/src/webapp-filtered/editor/ckeditor.launch.js:197]: i'm surprised that this fixes the issue? if portal is undefined, the previous code should work. has this been tested?

Comment (danielmerino) [reply library/src/webapp-filtered/editor/ckeditor.launch.js:197]: [Looking in StackOverflow](https://stackoverflow.com/a/3390426) it seems that when the variable hasn't been declared it throws a ReferenceError using the first check.

I don't know why Evalsys is failing, in Sakai 20 this didn't happen. There is not a QA Server with Evalsys where I could test this patch. But I have tested it locally and this change fixes our issue.

Comment (adrianfish) [reply library/src/webapp-filtered/editor/ckeditor.launch.js:197]: Dan's right. I just read that SO and it makes sense. You could also do this:
window?.portal?.siteId

window is guaranteed to be defined, so optional chaining will do the job here.

Comment (adrianfish) [inline library/src/webapp-filtered/editor/ckeditor.launch.js:197]:
```suggestion
        if (collectionId.startsWith('/user/') && window?.portal?.siteId) {
```

**Review by adrianfish — CHANGES_REQUESTED** [review]
You're right, Dan, about the referenceerror. Nice catch. You can use optional chaining though, instead of typeof.

**Comment by danielmerino** [comment]
Change done. I have tested here that it works too.

**Review by adrianfish — APPROVED** [review]
Thanks Dan

**Pull request merged**